src/pynwb/ophys.py

```diff
@@ -238,21 +238,40 @@ def __init__(self, **kwargs):
         self.reference_images = reference_images
 
     @docval({'name': 'pixel_mask', 'type': 'array_data', 'default': None,
-             'doc': 'pixel mask for 2D ROIs: [(x1, y1, weight1), (x2, y2, weight2), ...]',
+             'doc': ('Pixel mask for 2D ROIs: [(x1, y1, weight1), (x2, y2, weight2), ...]. '),
              'shape': (None, 3)},
             {'name': 'voxel_mask', 'type': 'array_data', 'default': None,
-             'doc': 'voxel mask for 3D ROIs: [(x1, y1, z1, weight1), (x2, y2, z2, weight2), ...]',
+             'doc': ('Voxel mask for 3D ROIs: [(x1, y1, z1, weight1), (x2, y2, z2, weight2), ...].'),
              'shape': (None, 4)},
             {'name': 'image_mask', 'type': 'array_data', 'default': None,
-             'doc': 'image with the same size of image where positive values mark this ROI',
+             'doc': ('Image with the same size of image where positive values mark the ROI weights for '
+                     'each pixel of the image.'),
              'shape': [[None]*2, [None]*3]},
             {'name': 'id', 'type': int, 'doc': 'the ID for the ROI', 'default': None},
             allow_extra=True)
     def add_roi(self, **kwargs):
-        """Add a Region Of Interest (ROI) data to this"""
+        """Add a Region Of Interest (ROI) data to this PlaneSegmentation table
+
+        Add an ROI using an image mask, a pixel mask, or a voxel mask. An image mask is an image with
+        the same size of the data image where positive values mark the ROI weights for each pixel of
+        the image. A pixel mask is a list of triplets [(x1, y1, weight1), (x2, y2, weight2), ...] or
+        Nx3 array that indicates the ROI weights for each pixel. Pixels without a weight have a
+        weight of 0. A voxel mask is a list of quadruplets or Nx4 array, that is similar to the pixel
+        mask, except the format is (x1, y1, z1, weight1).
+
+        If the NWB file will be compressed, then using an image mask will usually result in a smaller
```

Comment:
Are we suggesting that users compress the entire NWB file? Is there a way of just compressing the image masks?

Reply:
Good point. I will add 'data' to the allowed types for the masks and test this out.